exe/node/Main.hs

```diff
@@ -193,14 +193,19 @@ main = do
                                     ]
                                     (Telemetry.telemetryProbe telemetry)
 
+            -- When the node finished syncing for the first time, start the
+            -- miner and link the async to the main thread.
+            let onNodeReady _result = liftIO $
+                    link =<< async (miner node gossip)
```

Comment:
This will crash the syncing if the miner crashes. Are we ok with that? We also don’t get a handle on the miner thread. This might be an issue if we want to decouple it more.

Reply:
> This will crash the syncing if the miner crashes. Are we ok with that?

I suppose we don't have too much of an alternative, due to the way our spaghetti `Concurrently` is orchestrated at the moment. I think (but I might be wrong) but if any of the `Concurrently` dies, the other ones will follow, so this won't change (fundamentally) the disaster recovery strategy of our node.

> We also don’t get a handle on the miner thread. This might be an issue if we want to decouple it more.

Agreed, but I wouldn't try to get this perfect the first time, we can easily extend this further as we see the need for.